pkg/ring/util_test.go

```diff
@@ -184,3 +239,63 @@ func TestWaitRingStabilityShouldReturnErrorIfMaxWaitingIsReached(t *testing.T) {
 
 	assert.InDelta(t, maxWaiting, elapsedTime, float64(2*time.Second))
 }
+
+func TestWaitInstanceStateTimeout(t *testing.T) {
+	t.Parallel()
+
+	const (
+		instanceID      = "test"
+		timeoutDuration = time.Duration(3)
```

Comment:
I guess you wanted 3s and not 3ns. You should change this to `time.Duration(3 * time.Second)`. I think 1s is enough for this test, so I would just do `time.Second`.

_Same comment applies to other tests._